Library/Homebrew/download_strategy.rb

```diff
@@ -400,6 +400,8 @@ def fetch(timeout: nil)
         resolve_url_basename_time_file_size(url, timeout: end_time&.remaining!)
       # Authorization is no longer valid after redirects
       meta[:headers]&.delete_if { |header| header.start_with?("Authorization") } if is_redirection
+      # also apply to Casks
```

Comment:
Do casks not also support `headers` as well?

Reply:
we tried to use headers as in Formula but in Cask we got an error. Maybe one future feature could be to allow using multiple headers in casks as you can do in formula

Reply:
It may be that only `:header` is currently allowed in `Cask::DSL::URL`.

Reply:
done. Please review when you can.

Reply:
I've changed this PR to draft because the merge of the ":header" argument in the "initialize" method isn't working (the ":headers" are not being used by cURL to create the cURL parameters, so the URL is being invoked without an authorization header).

I will try to find a solution as soon as possible; otherwise, we will be forced to go back to the first approach.

Reply:
fixed!! I missed a call to super on initialize